ros/lib/rbac_interface.py

```diff
@@ -104,18 +105,23 @@ def ensure_has_permission(**kwargs):
             for p in perms:
                 if p in kwargs["permissions"]:
                     return  # allow
-            raise HTTPError(
+
+            # if wrong permissions
+            abort(
                 HTTPStatus.FORBIDDEN,
-                message="user does not have access to %s" % kwargs["permissions"],
+                message='user does not have access to %s' % kwargs["permissions"]
             )
-        except RBACDenied:
-            raise HTTPError(
+        # if something wrong inside try e.g. perms not of type array
+        except Exception:
```

Comment:
Here, you have added  `Exception` that means this block will get executed when there is an exception. Any specific reason that you have passed status code `HTTPStatus.FORBIDDEN` to abort helper method for any kind of exception?

Reply:
As you can see now, I have replaced the `Exception` with a custom exception called `RBACDenied`. And at this place, the exception raised should be related to access forbidden and hence, I added Forbidden.